core/field_dropdown : always return something.

LilyMakesThings opened the pull request:
### Resolves

This common error when a menu receives a completely empty array.
![image](https://github.com/TurboWarp/scratch-blocks/assets/127533508/3403c1c8-071a-4610-aca2-99dc5e97611d)

### Proposed Changes

Have getOptions always return an array with at least a blank string.

### Reason for Changes

This error is commonly found among following steps in the TurboWarp extension documentation. Especially for people working with the much less documented dynamic menus, you wouldn't expect such a common case to be the cause of this.

It's not entirely obvious or logical that this would be caused by an empty menu, especially when it should be handled internally.

### Test Coverage

None.

LilyMakesThings committed:
Allow variables to go anywhere

LilyMakesThings committed:
Merge branch 'TurboWarp:develop' into develop

LilyMakesThings committed:
Merge branch 'TurboWarp:develop' into develop

LilyMakesThings committed:
this upsets me more than it should

LilyMakesThings committed:
Merge branch 'TurboWarp:develop' into develop

LilyMakesThings committed:
Merge branch 'TurboWarp:develop' into develop

LilyMakesThings committed:
Update field_dropdown.js

LilyMakesThings committed:
Update field_dropdown.js

LilyMakesThings committed:
Update data.js

GarboMuffin commented on core/field_dropdown.js:
```diff
@@ -164,7 +164,7 @@ Blockly.FieldDropdown.prototype.init = function() {
 Blockly.FieldDropdown.prototype.showEditor_ = function() {
   var options = this.getOptions();
-  if (options.length == 0) return;
+  if (options.length == 0 || options == ['']) return;
```
comparing arrays like that will always be false

LilyMakesThings committed:
Compare arrays properly

Don't know which method is the most tasteful in this case so I just went with the one I already knew
